File: 3146. Permutation Difference between Two Strings.js

var findPermutationDifference = function(s, t) {
    let hMap = new Map();
    for(let i = 0 ; i < s.length ; i++){
        hMap.set(s[i], i);
    }
    let diff = 0;
    for(let i = 0 ; i < t.length ; i++){
        diff = diff + Math.abs(hMap.get(t[i]) - i)
    }
    return diff;
};

File: 3146. Permutation Difference between Two Strings.cs
public class Solution {
    public int FindPermutationDifference(string s, string t) {
        var dic = new Dictionary<char, int>();

        for(int i = 0 ; i < s.Length ; i++)
        {
            dic.Add(s[i], i);
        }
        int diff = 0;
        for(int j = 0 ; j < t.Length ; j++)
        {
            diff += Math.Abs(dic[t[j]] - j);
        }
        return diff;
    }
}